### PR TITLE
Update `rom-core` to >= 5.2.5

### PR DESCRIPTION
**Pull request by v-kolesnikov**
~Draft PR to test https://github.com/rom-rb/rom/pull/623~

### CODE REVIEW

File: rom-elasticsearch.gemspec [hunk 1]
```diff
@@ -18,7 +18,6 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
 
-  spec.add_runtime_dependency 'rom-core', '~> 5.2', '>= 5.2.4'
```
Comment (flash-gordon) [inline rom-elasticsearch.gemspec:21]: this must be updated, not removed

Comment (v-kolesnikov) [reply rom-elasticsearch.gemspec:21]: I thought rom-core is merged to rom in rom 6.0, isn't it?

Comment (flash-gordon) [reply rom-elasticsearch.gemspec:21]: yeah but adapters do depend on rom
https://github.com/rom-rb/rom-sql/blob/master/rom-sql.gemspec#L29

**Comment by solnic** [comment]
FWIW, 3.0.0 compat is now fixed in rom-core 5.2.5

**Comment by v-kolesnikov** [comment]
It is failed on coverage reporting step: https://github.com/rom-rb/rom-elasticsearch/pull/23/checks?check_run_id=1713124524#step:12:36
Looks like spec/support/coverage.rb is not required anywhere in specs.

**Comment by v-kolesnikov** [comment]
@solnic It finally works!

**Review by solnic — APPROVED** [review]
🎉

**Pull request merged**